Edit user profile

MichaelParadis opened the pull request:
Bennett and Kieter were fine with the UI all that is need is code review. I will take care of the merge conflicts if needed.

Hreherch committed:
Merge branch 'edit-user-profile' of https://github.com/CMPUT301F16T01/Carrier into edit-user-profile

MichaelParadis committed:
Made it so buttons respond the way I expect them to. Need to add it so the keyboard automatically hides.

MichaelParadis committed:
Added code that should work once we implement usercontroller more.

Hreherch committed:
Merge branch 'edit-user-profile' of https://github.com/CMPUT301F16T01/Carrier into edit-user-profile

Hreherch commented:
Issue: The text box appears when not in edit mode.
How to:
1. press edit button
2. edit some text
3. save or cancel
4. press the text box again, and the keyboard will reappear. (can't edit it)

This is no bueno. 🌮

MichaelParadis committed:
Only possible reason for the problem.

MichaelParadis committed:
Added more stuff that should fix the problem.

MichaelParadis commented:
@Hreherch I have no idea why that happens. I am unable to bring up the keyboard unless I am in the edit mode. Although I have encountered a  bug with the email edit text which allows the changing of the text while not in edit mode. This is done by:
1. edit email
2. cancel or save without closing the keyboard
3. Start typing with the keyboard still up.

Although I can replicate this bug there is no reason that I can find for this happening since it only happens with the email edit text. I am unable to replicate this when I try to edit the phone number text.

Hreherch committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into edit-user-profile

# Conflicts:
#	app/src/main/AndroidManifest.xml

Hreherch committed:
Removed several duplicatations of XML elements?

:(

Hreherch committed:
Fixed my OCD with this activity

Keyboards are not annoying, the EditTexts now respond only when editing.
So much QoL for me, lol.

Hreherch committed:
Merge remote-tracking branch 'origin/edit-user-profile' into edit-user-profile

# Conflicts:
#	app/src/main/res/layout/activity_user_profile.xml

Apparently Disk and Main Memory did not like each other for a second and
I had a conflict with myself?

Hreherch commented:
I have fixed the merge conflict with master.

I has also fixed the issues I had with the UI. Note that I had to remove the highlighting of the EditText, but I made a cursor workaround that should be okay for now. 

I will say that I believe the code can be cleaned up and generalized, as there is a lot of functions that do very similar things. At the very least, there should be at most three (edit, save, delete) functions and at best two (edit, save+delete). Right now there are six different functions and editing one means editing at least one other one... this is a low priority issue. @kieter and @MichaelParadis did a good job getting this initialized. 

Furthermore, please immediately fix a merge conflict when you make a pull request in the future. 

**I approve this branch to be merged with master.** Someone else can do it, they don't need to check it over but this allows them to.

kieter commented on app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java:
```diff
@@ -1,38 +1,73 @@
-/**
- * Created by kiete on 10/18/2016.
- */
+// TODO A lot can be done to reduce code duplication. Notice how there are 4 functions that close,
```
I concur

kieter commented:
@MichaelParadis please review the changes Bennett made in this pull request, they're good 👍 .

kieter commented on app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java:
```diff
@@ -108,10 +154,14 @@ public void editEmailAddress(View v){
         EditText emailView = (EditText) findViewById(R.id.EmailEditText);
         // Set it so the user can edit the EditText
-        emailView.setFocusable(true);
+        emailView.setFocusableInTouchMode(true);
```
I just looked this up. It's interesting, what "Focus" and "FocusInTouchMode" is.

kieter commented on app/src/main/java/comcmput301f16t01/github/carrier/UserProfileActivity.java:
```diff
@@ -1,38 +1,73 @@
+        String username;
+
+        // Developer code, may not be needed for release...
+        if (currentUser == null) {
```
This is pretty handy.
